### PR TITLE
Request source from server when path is missing

### PR DESCRIPTION
**Pull request by timfel**

### CODE REVIEW

**Comment by timfel** [comment]
I've squashed the commits. I still don't understand if the CI is failing because of something I did though

**Comment by yyoncho** [comment]
CI failure is not related to your change.

**Comment by timfel** [comment]
Friendly ping 🙂 Any feedback or improvement requests?

**Comment by yyoncho** [comment]
Sorry for the delay. The code looks good but I will have to manually test with a few adapters. I will try to do it during the weekend.

File: dap-mode.el [hunk 1]
```diff
@@ -763,6 +763,28 @@ will be reversed."
         (dap-debug (dap--debug-session-launch-args debug-session)))
     (user-error "There is session to restart")))
 
+(defun dap--request-source-for-frame-by-reference (debug-session stack-frame line column name)
+  "Request source string for a STACK-FRAME using the sourceReference."
+  (with-lsp-workspace (dap--debug-session-workspace debug-session)
```
Comment (yyoncho) [inline dap-mode.el:768]: with-lsp-workspace seems like it is redundant.

**Review by yyoncho — COMMENTED** [review]
Looks good, one minor comment.

**Pull request merged**